lib/recurly/risk/three-d-secure/strategy/braintree.js

```diff
@@ -0,0 +1,96 @@
+import loadScript from 'load-script';
+import Promise from 'promise';
+import ThreeDSecureStrategy from './strategy';
+
+const debug = require('debug')('recurly:risk:three-d-secure:braintree');
+const BRAINTREE_CLIENT_VERSION = '3.50.0';
+
+export default class BraintreeStrategy extends ThreeDSecureStrategy {
+
+  static strategyName = 'braintree_blue';
+
+  constructor (...args) {
+    super(...args);
+
+    debug('loading braintree libraries');
+    this.loadBraintreeLibraries()
+      .catch(cause => this.threeDSecure.error('3ds-vendor-load-error', { vendor: 'Braintree', cause }))
+      .then(() => {
+        this.braintree = window.braintree;
+        debug('Braintree checkout instance created', this.braintree);
+        this.markReady();
+      });
+  }
+
+  get braintreeClientToken () {
+    return this.actionToken.three_d_secure.params.client_token;
+  }
+
+  get nonce () {
+    return this.actionToken.three_d_secure.params.payment_method_nonce;
+  }
+
+  get amount () {
+    return this.actionToken.transaction.amount_in_cents;
+  }
+
+  get bin () {
+    return this.actionToken.three_d_secure.params.bin;
+  }
+
+  /**
+   * Provides the target DOM element for which we will apply
+   * fingerprint detection, challenge flows, and results
+   *
+   * @param {HTMLElement} element
+   */
+  attach (element) {
+    super.attach(element);
+    let braintreeThreeDSecure;
+
+    debug('Attempting to load braintree');
+    this.whenReady(() => {
```

Comment:
To make things a bit easier to read through in this callback, we can do some destructuring assignment.

```js
const { braintree, braintreeClientToken, amount, nonce, bin, threeDSecure } = this;
```

Then you'll be able to remove all of the `this` references.